xcode/Subconscious/Shared/Components/Detail/UserProfileDetailView.swift

```diff
@@ -501,16 +482,26 @@ struct UserProfileDetailModel: ModelProtocol {
             )
             
         case .attemptUnfollow:
-            guard let did = state.unfollowCandidate?.did else {
+            guard let candidate = state.unfollowCandidate else {
                 return Update(state: state)
             }
             
+            let x = Func.run {
+                switch (candidate.ourFollowStatus) {
+                case .following(let name):
+                    return environment.addressBook
+                        .unfollowUserPublisher(
+                            did: candidate.did,
+                            petname: name
```

Comment:
Not sure this makes sense. We want to unfollow the user by checking if the leaf name of their address is the same as one of our petnames for them and unfollow _that_ name, preserving any others.

If the leaf name and our petname do not match, then unfollow by DID instead pursing all names.